wagtail/wagtailsearch/backends/elasticsearch2.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import absolute_import, unicode_literals
+
+from wagtail.wagtailsearch.index import FilterField, RelatedFields, SearchField
+
+from .elasticsearch import (
+    ElasticsearchIndex, ElasticsearchMapping, ElasticsearchSearchBackend, ElasticsearchSearchQuery,
+    ElasticsearchSearchResults)
+
+
+def get_model_root(model):
+    """
+    This function finds the root model for any given model. The root model is
+    the highest concrete model that it descends from. If the model doesn't
+    descend from another concrete model then the model is it's own root model so
+    it is returned.
+
+    Examples:
+    >>> get_model_root(wagtailcore.Page)
+    wagtailcore.Page
+
+    >>> get_model_root(myapp.HomePage)
+    wagtailcore.Page
+
+    >>> get_model_root(wagtailimages.Image)
+    wagtailimages.Image
+    """
+    if model._meta.parents:
+        parent_model = list(model._meta.parents.items())[0][0]
```

Comment:
I see, thanks for the suggestion though!

On 23 August 2016 at 11:26, Mikalai Radchuk <notifications@github.com>
wrote:

> In wagtail/wagtailsearch/backends/elasticsearch2.py
> <https://github.com/torchbox/wagtail/pull/2573#discussion_r75839517>:
>
> > +    the highest concrete model that it descends from. If the model doesn't
> > +    descend from another concrete model then the model is it's own root model so
> > +    it is returned.
> > +
> > +    Examples:
> > +    >>> get_model_root(wagtailcore.Page)
> > +    wagtailcore.Page
> > +
> > +    >>> get_model_root(myapp.HomePage)
> > +    wagtailcore.Page
> > +
> > +    >>> get_model_root(wagtailimages.Image)
> > +    wagtailimages.Image
> > +    """
> > +    if model._meta.parents:
> > +        parent_model = list(model._meta.parents.items())[0][0]
>
> It works for Python 2, but not for 3, sorry.
>
> Python 2:
>
> >>> from collections import OrderedDict>>> xy = OrderedDict([('x', None), ('y', None)])>>> xy.keys()[0]'x'
>
> Python 3:
>
> >>> from collections import OrderedDict>>> xy = OrderedDict([('x', None), ('y', None)])>>> xy.keys()[0]
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>TypeError: 'odict_keys' object does not support indexing>>> list(xy.keys())[0]'x'
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/torchbox/wagtail/pull/2573/files/39a8e81ec68d2dcf3ec949ce13c576f06b5eb532#r75839517>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABCwsMrjEyIrug54RRT1IAUu84vWumKVks5qisrGgaJpZM4IY7zC>
> .
>